applicationconfig/client/README.md

```diff
@@ -0,0 +1,95 @@
+Azure App Configuration is a managed service that helps developers centralize their application configurations simply and securely.
+
+Use the client library for App Configuration to:
+- Add a new setting configuration
+- Update an existing setting configuration
+- Retrieve a setting configuration
+- Delete a setting configuration
+
+[Source code]() | [Package (Maven)]() | [API reference documentation]() | [Product documentation](https://docs.microsoft.com/en-us/azure/azure-app-configuration/)
+
+## Getting started
+
+### Prerequisites
+
+- JDK 8 or later
+- Apache Maven 3 or above
+- Azure Service Principal
+- App Configuration Store
+
+### Adding the package to your product
+
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-app-configuration</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+</dependency>
+```
+
+### Authenticate the client
+
+```java
+String connectionString = "endpoint={endpoint_value};id={id_value};secret={secret_value}";
+ConfigurationClient client = ConfigurationClient.builder()
+        .credentials(new ConfigurationClientCredentials(connectionString)
+        .build();
+```
+
+Using the Azure web portal to open your App Configuration Store's Access Key section, choose one of the two connection strings and use it to replace the string value above.
+
+## Key concepts
+
+### [Key-value store](https://docs.microsoft.com/en-us/azure/azure-app-configuration/concept-key-value)
+
+Configurations are stored using key-value pairs. Keys must be unique for each configuration so it's common practice to organize keys into hierarchical namespaces using delimiting characters, for example AppName:Service1:DB:Endpoint. Optionally a label attribute may be given to a key to differentiate two configurations, for example having different versions of the configuration. Values are stored as unicode strings.
+
+### [Point-in-time snapshot](https://docs.microsoft.com/en-us/azure/azure-app-configuration/concept-point-time-snapshot)
+
+When a configuration is modified a point-in-time snapshot is recorded, snapshots are retained for seven days.
+
+### Responses
+
+```Java
+public class ConfigurationSetting {
+    private String key;
+    private String label;
+    private String value;
+    private String contentType;
+    private String etag;
+    private OffsetDateTime lastModified;
+    private boolean locked;
+    private Map<String, String> tags;
+}
+```
+
+Successful requests to the service will have a response containing a ConfigurationSetting as the body. ConfigurationSetting contains the key-value and metadata about the configuration.
+
+An unsuccessful request will throw an error that indicates why the request failed. (todo - update this with further error type explanation when implemented)
+
+For further information about responses view the [Rest API reference](https://github.com/Azure/AppConfiguration#rest-api-reference).
+
+## Examples
+
+### Create a new setting configuration
```

Comment:
Is the idea to add these examples in future PR?

Reply:
Yup, @conniey is creating samples in PR #3288, after that is merged in we can take some of the code snippets from it and put them in the README

Reply:
Ohh right!